### ISSUE TITLE
HP Pavillion Laptop only works on Legacy Mode

### ISSUE THREAD

**Issue by migue802** [Metabolix/HackBGRT#160]
### Problem

As stated before in https://github.com/Metabolix/HackBGRT/issues/77#issuecomment-1790395756, my HP Pavillion x360 laptop won't show load the HackBGRT EFI file before the windows original EFI file.

Here's the setup.log file with some comments on it: [setup.log](https://github.com/Metabolix/HackBGRT/files/13457349/setup.log)
~_I've tried to paste the logs here on a spoiler but the body is too long xd_~

Please note my system is in Spanish, as i've seen that is an issue detecting BitLocker, so it might be an issue also for other things. Let me know if I should change the language to english to temporaly install HackBGRT.

### System details
System Model: HP Pavilion x360 Convertible 14-cd0xxx
OS: Windows 11
OS Version: 10.0.22621 compilation 22621
HackBGRT Ver.: 2.2.0

Please let me know if I need to debug something else.

**Comment by migue802** [comment]
Also to note: I've disabled Secure Boot, Hibernation is enabled but disabling it doesn't make any difference yet

**Comment by Metabolix** [comment]
Thanks, this is very interesting!

The first method (enable-entry) seems to write and enable a boot entry (numbered Boot2002), but apparently this gets erased. (Boot20## seems to be a common choice for firmware-generated entries, so maybe I should use only Boot00## values. The UEFI spec doesn't say anything about reserved values, though.)

The second method (enable-bcdedit) creates the correct boot entry (numbered Boot0000 this time) but apparently `/addfirst` fails to actually make it the default choice, even though it says it succeeded. I will try to debug this later.

Based on the logs, could you please try this sequence:
1. Uninstall.
2. Install with the second method (enable-bcdedit).
3. Install with the first method (enable-entry).
4. Reboot.

**Comment by migue802** [comment]
Hey! Will do that, let me a sec. Thanks for the interest tho, this is a very cool program and it would be cool if it works for all computers

**Comment by migue802** [comment]
Yes!!! It worked perfectly; I've noticed a bit longer loading screen but it's okay as it's worth looking at the custom logo XD

**Comment by migue802** [comment]
Hey @Metabolix, I was investigating the issue and I thought localization/system language has something to do with this issue, and I was maybe right? 
I don't know how but I switched the account language to English (United States), uninstalled and installed with regular method and it kinda worked for some reason? I just had to set up the boot order and that's why I switched back to Spanish and for some reason, if I uninstall and do regular installation it perfectly works, it's just a weird behaviour but here's the updated [setup.log](https://github.com/Metabolix/HackBGRT/files/13472915/setup.log) if you want to take a look to it. 
What a weird issue, isn't it?

**Comment by migue802** [comment]
Update: Spanish doesn't work xd, it just works if I set my system language to English (United States)

**Comment by migue802** [comment]
~Please ignore all the deleted comments, said things without testing and doing it multiple times, silly me... :p~

**Comment by Metabolix** [comment]
Thanks for testing!

In the new log it looks like `bcdedit` manages to change the boot order after all, so it's not clear why it didn't work earlier.

I've now published v2.3.0 where I have fixed the boot entry numbering (in the default installation method) to avoid collisions with any firmware-specific entries. I think this should fix the original problem (although I can't know for sure).

> ~Please ignore all the deleted comments,

I read them anyway, and for the record, this issue doesn't have anything to do with language settings. Language should affect only BitLocker detection (which then requests user input) and some parts of automatic ESP detection (which then uses other tests to find the drive).
